packages/nx/src/command-line/release/meta.ts

```diff
@@ -0,0 +1,102 @@
+import { prompt } from 'enquirer';
+import { readNxJson } from '../../config/nx-json';
+import { createProjectGraphAsync } from '../../project-graph/project-graph';
+import { handleErrors } from '../../utils/params';
+import { releaseChangelog } from './changelog';
+import { MetaOptions, VersionOptions } from './command-object';
+import {
+  NxReleaseConfig,
+  createNxReleaseConfig,
+  handleNxReleaseConfigError,
+} from './config/config';
+import { releasePublish } from './publish';
+import { NxReleaseVersionResult, releaseVersion } from './version';
+
+export const releaseMetaCLIHandler = (args: VersionOptions) =>
+  handleErrors(args.verbose, () => releaseMeta(args));
+
+export async function releaseMeta(
+  args: MetaOptions
+): Promise<NxReleaseVersionResult | number> {
+  const nxReleaseConfig = await readNxReleaseConfig(args);
+
+  const versionResult: NxReleaseVersionResult = await releaseVersion({
+    ...args,
+    // if enabled, committing and tagging will be handled by the changelog
+    // command, so we should only stage the changes in the version command
+    stageChanges: args.gitCommit || nxReleaseConfig.git?.commit,
+    gitCommit: false,
```

Comment:
I think we need some more config validation before proceeding.

If they have explicitly put `version.git.commit` and/or `version.git.tag` in their config, we should error before attempting to run the meta command (rather than just setting things to false here like you have) because it doesn't make sense.

There are other such permutations as well when it comes to changelog.

Therefore I think the simplest and clearest thing would be, any `version.git` or `changelog.git` config set on the user side is incompatible with the meta command and we hard error up front explaining that.

Root level `git` is naturally fine (and you are already appropriately leveraging it on line 27)